nomt/src/beatree/ops/update/leaf_updater.rs

```diff
@@ -14,17 +14,61 @@ use super::{
 
 pub struct BaseLeaf {
     pub node: LeafNode,
-    pub iter_pos: usize,
     pub separator: Key,
+    low: usize,
 }
 
 impl BaseLeaf {
-    fn next_key(&self) -> Option<Key> {
-        if self.iter_pos >= self.node.n() as usize {
-            None
-        } else {
-            Some(self.key(self.iter_pos))
+    pub fn new(node: LeafNode, separator: Key) -> Self {
+        BaseLeaf {
+            node,
+            separator,
+            low: 0,
+        }
+    }
+
+    // Try to find the given key starting from `self.low` up to the end.
+    // Returns whether the key is present or not and the index of the key
+    // or the index containing the first key bigger then the one specified.
+    fn find_key(&mut self, key: &Key) -> Option<(bool, usize)> {
```

Comment:
wondering if there is a way to simplify and just use the std binary search functions?

Reply:
Yes, we've discussed it in the scenario of having a slice or something from where a slice could be extracted easily. Initially implementing it given the layout of the leaf node, I thought that it would have been easier to just use `self.key(i)` and implement the binary search.

After reviewing it now, I notice how `cell_pointers` could be extracted once at the beginning and then call `binary_search_by` on it. The new code will be part of a follow-up pr!